# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/firefox/hotfix-simd-sse2.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

# Backport of https://github.com/simd-everywhere/simde/commit/fee149ae76155f05fbae921884afe68260f2ff83

236:47.04 In file included from rlboxsoundtouch.wasm.c:706:
236:47.04 /usr/src/t2-src/src.firefox.p3.250410.000710.24457/firefox-137.0/third_party/simde/simde/wasm/simd128.h: In function 'simde_wasm_f32x4_min':
236:47.05 /usr/src/t2-src/src.firefox.p3.250410.000710.24457/firefox-137.0/third_party/simde/simde/wasm/simd128.h:4741:12: error: 'simde_v128_private'
 has no member named 'sse_m128i'; did you mean 'sse_m128'?

--- firefox-137.0/third_party/simde/simde/wasm/simd128.h.vanilla	2025-04-10 09:57:27.585287495 +0200
+++ firefox-137.0/third_party/simde/simde/wasm/simd128.h	2025-04-10 10:00:23.445699248 +0200
@@ -4728,7 +4728,7 @@
       b_ = simde_v128_to_private(b),
       r_;
 
-    #if defined(SIMDE_X86_SSE_NATIVE)
+    #if defined(SIMDE_X86_SSE2_NATIVE)
         // Inspired by https://github.com/v8/v8/blob/c750b6c85bd1ad1d27f7acc1812165f465515144/src/codegen/shared-ia32-x64/macro-assembler-shared-ia32-x64.cc#L202
         simde_v128_private scratch;
         scratch.sse_m128 = a_.sse_m128;
@@ -4765,7 +4765,7 @@
       b_ = simde_v128_to_private(b),
       r_;
 
-    #if defined(SIMDE_X86_SSE_NATIVE)
+    #if defined(SIMDE_X86_SSE2_NATIVE)
         // Inspired by https://github.com/v8/v8/blob/c750b6c85bd1ad1d27f7acc1812165f465515144/src/codegen/shared-ia32-x64/macro-assembler-shared-ia32-x64.cc#L263
         simde_v128_private scratch;
         scratch.sse_m128d = a_.sse_m128d;
@@ -5025,7 +5025,7 @@
       b_ = simde_v128_to_private(b),
       r_;
 
-    #if defined(SIMDE_X86_SSE_NATIVE)
+    #if defined(SIMDE_X86_SSE2_NATIVE)
         // Inspired by https://github.com/v8/v8/blob/c750b6c85bd1ad1d27f7acc1812165f465515144/src/codegen/shared-ia32-x64/macro-assembler-shared-ia32-x64.cc#L231
         simde_v128_private scratch;
         scratch.sse_m128 = a_.sse_m128;
@@ -5063,7 +5063,7 @@
       b_ = simde_v128_to_private(b),
       r_;
 
-    #if defined(SIMDE_X86_SSE_NATIVE)
+    #if defined(SIMDE_X86_SSE2_NATIVE)
         // Inspired by https://github.com/v8/v8/blob/c750b6c85bd1ad1d27f7acc1812165f465515144/src/codegen/shared-ia32-x64/macro-assembler-shared-ia32-x64.cc#L301
         simde_v128_private scratch;
         scratch.sse_m128d = a_.sse_m128d;
@@ -8747,7 +8747,7 @@
       a_ = simde_v128_to_private(a),
       r_;
 
-    #if defined(SIMDE_X86_SSE_NATIVE)
+    #if defined(SIMDE_X86_SSE2_NATIVE)
       r_.sse_m128d = _mm_sqrt_pd(a_.sse_m128d);
     #elif defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vsqrtq_f64(a_.neon_f64);
